src/client/datascience/kernel-launcher/jupyterPaths.ts

```diff
@@ -76,27 +104,41 @@ export class JupyterPaths {
      * We need to look at the 'kernels' sub-directory and these paths are supposed to come first in the searching
      * https://jupyter.readthedocs.io/en/latest/projects/jupyter-directories.html#envvar-JUPYTER_PATH
      */
+    @traceDecorators.verbose('Get Jupyter Paths')
     private async getJupyterPathPaths(cancelToken?: CancellationToken): Promise<string[]> {
```

Comment:
The logging and caching was very useful, see comments earlier.

Reply:
Same comment on this one. I swear you had something that would cache results for a function call into a memento.

Reply:
Yes now I remember, we had code to cache in memento, I think its in the Python extension, & it may have been removed from this repo as it wasn't used.